api/iteration.go

```diff
@@ -73,3 +83,28 @@ func (iter *Iteration) isValidFilepath(path string) bool {
 	}
 	return strings.HasPrefix(strings.ToLower(path), strings.ToLower(iter.Dir))
 }
+
+func readFileAsUTF8String(filename string) (*string, error) {
+	b, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	encoding, _, _ := charset.DetermineEncoding(b, mimeType)
+	decoder := encoding.NewDecoder()
+	decodedBytes, _, err := transform.Bytes(decoder, b)
+	if err != nil {
+		return nil, err
+	}
+
+	// Drop the UTF-8 BOM that may have been added. This isn't necessary, and
+	// it's going to be written into another UTF-8 buffer anyway once it's JSON
+	// serialized.
+	//
+	// The standard recommends omitting the BOM. See
+	// http://www.unicode.org/versions/Unicode5.0.0/ch02.pdf
+	decodedBytes = bytes.TrimPrefix(decodedBytes, utf8BOM)
+
+	s := string(decodedBytes)
+	return &s, nil
```

Comment:
A habit I picked up from the style guidelines of another project. The idea is that any function should either return a pointer to an initialized, usable object or a nil pointer and an error, but never an uninitialized object.

I don't feel strongly about this. Just habit.